tests/gpflow/utilities/test_deepcopy.py

```diff
@@ -36,9 +40,18 @@ def test_deepcopy_component_clears_bijector_cache_and_deecopy(module):
     input = 1.0
     _ = module(input)
     with pytest.raises(TypeError):
-        deepcopy(module)
-    module_copy = deepcopy_components(module)
+        copy.deepcopy(module)
+    module_copy = deepcopy(module)
     assert module.var == module_copy.var
     assert module.var is not module_copy.var
     module_copy.var.assign([5.0])
     assert module.var != module_copy.var
+
```

Comment:
can we have some tests for frozen deepcopy with lists, nested lists, dictionaries etc? I realise that's a lot of tests but there's a lot that can go wrong with this kind of functionality

Reply:
perhaps this can be done efficiently with parametrised tests